Refreshed documentation so it reflects how users should interact with… (#266)
casper-network/casper-sidecar: rpc_sidecar/README.md

## Synopsis

> The rpc-sidecar is a process that is connected to binary RPC port of a casper node and exposes a JSON-RPC interface for interacting with that node. The RPC protocol allows for basic operations like querying global state, sending transactions and deploys etc. All of the RPC methods are documented [here](https://docs.casper.network/developers/json-rpc/).

Reviewer: ```suggestion
The Casper Event Sidecar is a process that connects to the RPC port of a Casper node and exposes a JSON-RPC interface for interacting with that node. The RPC protocol allows for basic operations like querying global state, sending transactions and deploys, etc. All of the RPC methods are documented [here](https://docs.casper.network/developers/json-rpc/).
```

    - request to submit a transaction for execution
    - request to speculatively execute a transaction 

> The node does not interpret the data it sends where it's not necessary. For example, most database items are sent as opaque byte arrays and the sidecar is responsible for interpreting them. This leaves the sidecar in control of the data it receives and allows it to be more flexible in how it handles it.

Reviewer: Is this not applicable anymore?

Author: Yeah, actually some of the validation was moved to node, because this decoupling led to a very sub-optimal and talkative back-and-forth between sidecar and the node. So it surely is not true for all the endpoints.

> ## Discovering JSON RPC API
> 
> Once running, the rpc-sidecar can be queried for its API using the `rpc.discover` method.  This will return a list of all RPC methods and their parameters. You can query the server using the following command:

Reviewer: ```suggestion
## Discovering the JSON RPC API

Once running, the Sidecar can be queried for its JSON RPC API using the `rpc.discover` method, as shown below. The result will be a list of RPC methods and their parameters.
```
